checkstyle/src/main.rs

```diff
@@ -0,0 +1,124 @@
+use regex::Regex;
+use std::fs::{self};
+
+fn get_file(filename: &str) -> String {
+    fs::read_to_string(filename).expect("Failed to read file")
+}
+
+fn split_into_lines(s: &str) -> Vec<&str> {
+    s.lines().collect()
+}
+
+fn check_matching_regex_line(file: &str, re: Regex) -> Option<bool> {
+    let file_lines = split_into_lines(file);
+    for file_line in file_lines {
+        if re.is_match(file_line) {
+            return Some(true);
+        }
+    }
+    Some(false)
+}
+
+fn check_matching(file: &str, regex: &str) -> Option<bool> {
+    let lines: Vec<&str> = split_into_lines(regex);
+    for regex_line in lines {
+        let re = Regex::new(regex_line).ok()?;
+        if check_matching_regex_line(file, re) == Some(false) {
+            return Some(false);
+        }
+    }
+    Some(true)
+}
+
+fn list_files_in_folder(folder_path: &str) -> Result<Vec<String>, std::io::Error> {
+    let mut file_paths = Vec::new();
+
+    for entry in fs::read_dir(folder_path)? {
+        let entry = entry?;
+        let path = entry.path();
+
+        if path.is_file() {
+            if let Some(file_name) = path.file_name() {
+                if let Some(file_name_str) = file_name.to_str() {
+                    file_paths.push(file_name_str.to_string());
+                }
+            }
+        }
+    }
+
+    Ok(file_paths)
+}
+
+fn all_files(folder_path: &str) -> Vec<String> {
+    match list_files_in_folder(folder_path) {
+        Ok(file_paths) => {
+            return file_paths;
+        }
+        _ => {
+            return Vec::new();
+        }
+    }
```

Comment:
This is a manual `.unwrap_or_default()`:
```suggestion
   list_files_in_folder(folder_path).unwrap_or_default()
```